### PR TITLE
Puppeteer: fix test when workspaces count is zero

### PR DESCRIPTION
**Pull request by aweng98**
I created a new script that clean up (delete) `puppetmaster@fake-research-aou.org` user's workspaces. Tests can fail when user has 0 workspace.

### CODE REVIEW

File: e2e/app/workspace-card.ts [hunk 1]
```diff
@@ -31,10 +31,11 @@ export default class WorkspaceCard extends BaseElement {
     return resourceCards;
   }
 
-  static async getAnyCard(page: Page): Promise<WorkspaceCard> {
+  static async getAnyCard(page: Page): Promise<WorkspaceCard | null> {
```
Comment (gjuggler) [inline e2e/app/workspace-card.ts]: I don't think this change makes sense. An async function should only ever resolve the Promise if the expected return value is there; in other cases, an error should be returned so the Promise is rejected.

I also can't quite tell why this change is being made – all other changes in this PR are changing how the response to getAllCards is handled, not getAnyCard. Is there a particular test case which required this change?

File: e2e/tests/home/home-page.spec.ts [hunk 2]
```diff
@@ -32,6 +32,11 @@ describe('Home page ui tests', () => {
     await GoogleLoginPage.logIn(page);
 
     const cards = await WorkspaceCard.getAllCards(page);
+    if (cards.length === 0) {
```
Comment (gjuggler) [inline e2e/tests/home/home-page.spec.ts]: It feels very odd for a test to short-circuit depending on the structure of the page. If we believe that the assertions further down in the test case are important to effectively QA the app, then this early return should be considered a test breakage!

I can think of a couple options here:

1) As part of the set-up of this overall test file, include a call which creates at least one workspace.
2) Change the cleanup script to leave at least one workspace, so we never have to early-return from this test case.

File: e2e/tests/home/home-page.spec.ts [hunk 3]
```diff
@@ -98,7 +103,11 @@ describe('Home page ui tests', () => {
   test('Check Workspace card on Home page', async () => {
     await GoogleLoginPage.logIn(page);
 
-    await WorkspaceCard.getAllCards(page);
+    const cards = await WorkspaceCard.getAllCards(page);
```
Comment (gjuggler) [inline e2e/tests/home/home-page.spec.ts]: Same critique as above (and applies equally to all of the below early returns): I think it's incorrect to be returning early from a test, and skipping downstream calls to `expect`.

**Comment by aweng98** [comment]
@gjuggler thank you. I am closing this PR because I don't have a better and clean solution at this time. need more time to think about it.

**Pull request closed without merging**